deploy/ssh.sh

```diff
@@ -0,0 +1,205 @@
+  # If provided then CA intermediate certificate will be copied to provided filename.
+  if [ -n "$ACME_DEPLOY_SSH_CAFILE" ]; then
+    Le_Deploy_ssh_cafile="$ACME_DEPLOY_SSH_CAFILE"
+    _savedomainconf Le_Deploy_ssh_cafile "$Le_Deploy_ssh_cafile"
+  fi
+  if [ -n "$Le_Deploy_ssh_cafile" ]; then
+    # backup file we are about to overwrite.
+    _cmdstr="$_cmdstr cp $Le_Deploy_ssh_cafile $_backupdir ;"
+    # copy new certificate into file.
+    _cmdstr="$_cmdstr echo \"$(cat "$_cca")\" > $Le_Deploy_ssh_cafile ;"
+    _info "will copy CA file to remote file $Le_Deploy_ssh_cafile"
+  fi
+
+  # FULLCHAIN is optional.
+  # If provided then fullchain certificate will be copied to provided filename.
+  if [ -n "$ACME_DEPLOY_SSH_FULLCHAIN" ]; then
+    Le_Deploy_ssh_fullchain="$ACME_DEPLOY_SSH_FULLCHAIN"
+    _savedomainconf Le_Deploy_ssh_fullchain "$Le_Deploy_ssh_fullchain"
+  fi
+  if [ -n "$Le_Deploy_ssh_fullchain" ]; then
+    # backup file we are about to overwrite.
+    _cmdstr="$_cmdstr cp $Le_Deploy_ssh_fullchain $_backupdir ;"
+    # copy new certificate into file.
+    _cmdstr="$_cmdstr echo \"$(cat "$_cfullchain")\" > $Le_Deploy_ssh_fullchain ;"
+    _info "will copy full chain to remote file $Le_Deploy_ssh_fullchain"
+  fi
+
+  # REMOTE_CMD is optional.
+  # If provided then this command will be executed on remote host.
+  # A 2 second delay is inserted to allow system to stabalize after
+  # executing a service stop.
+  if [ -n "$ACME_DEPLOY_SSH_REMOTE_CMD" ]; then
+    Le_Deploy_ssh_remote_cmd="$ACME_DEPLOY_SSH_REMOTE_CMD"
+    _savedomainconf Le_Deploy_ssh_remote_cmd "$Le_Deploy_ssh_remote_cmd"
+  fi
+  if [ -n "$Le_Deploy_ssh_remote_cmd" ]; then
+    if [ -n "$Le_Deploy_ssh_service_stop" ]; then
+      _cmdstr="$_cmdstr sleep 2 ;"
+    fi
+    _cmdstr="$_cmdstr $Le_Deploy_ssh_remote_cmd ;"
+    _info "Will execute remote command $Le_Deploy_ssh_remote_cmd"
+  fi
+
+  # SERVICE_START is optional.
+  # If provided then this command will be executed on remote host.
+  # A 2 second delay is inserted to allow system to stabalize after
+  # executing a service stop or previous command.
+  if [ -n "$ACME_DEPLOY_SSH_SERVICE_START" ]; then
+    Le_Deploy_ssh_service_start="$ACME_DEPLOY_SSH_SERVICE_START"
+    _savedomainconf Le_Deploy_ssh_service_start "$Le_Deploy_ssh_service_start"
+  fi
+  if [ -n "$Le_Deploy_ssh_service_start" ]; then
+    if [ -n "$Le_Deploy_ssh_service_stop" ] || [ -n "$Le_Deploy_ssh_remote_cmd" ]; then
+      _cmdstr="$_cmdstr sleep 2 ;"
+    fi
+    _cmdstr="$_cmdstr $Le_Deploy_ssh_service_start ;"
+    _info "Will start remote service with command $Le_Deploy_ssh_remote_cmd"
+  fi
+
+  if [ -z "$_cmdstr" ]; then
+    _err "No remote commands to excute. Failed to deploy certificates to remote server"
+    return 1
+  else
+    # something to execute.
+    # run cleanup on the backup directory, erase all older than 180 days.
+    _cmdstr="find $_backupprefix* -type d -mtime +180 2>/dev/null | xargs rm -rf ; $_cmdstr"
```

Comment:
I have investigated and I still don't like it.  All we do is substitute two other commands in place of find and xargs.... date and expr and it becomes much much more complex.  This is what I came up with...
```
now=$(date -u +%s);
for fn in $_backupprefix*; do
  if [ "$(expr $(expr $now - $(date -ur $fn +%s)) / 86400)" -ge "180" ]; then
    rm -rf $fn
  fi
done
```
I don't know how portable this is.  And I am not testing for directories only.  And I am not handling the case where there are no files matching $_backupprefix* (in which case fn is set to $_backupprefix* which is not a valid filename / directory) so that would add additional complication.  I could simplify the math by doing -ge "15552000" (180 days * 86000 seconds) but that is crystal clear, right!

I much prefer the find/xargs version and feel like it is perfectly okay to document the requirements for the remote host.